src/tasks.py

```diff
@@ -342,3 +359,216 @@ def get_metric():
+        
+        x_k_all = torch.zeros(u_k.size()[0], u_k.size()[1], u_k.size()[2], device=u_k.device)
+        x_k_all[:, 0, :] = x_k_1[:, 0, :]
+
+        for i in range(u_k.size()[1] - 1):
+            x_k_all[:, (i+1):(i + 2), :] = x_k_all[:, i:(i + 1), :] @ A + u_k[:, (i + 1):(i + 2), :] @ B
+        
+        y_k = C @ torch.transpose(x_k_all, 1, 2)
+        y_k = torch.transpose(y_k, 1, 2)
+
+        #BELOW IS THE SCALING
+        y_k = (1/math.sqrt(len(y_k[0, :, 0]))) * y_k
+        return y_k[:, :, 0]
+
+    @staticmethod
+    def generate_rand_stable(batch_size, rows, cols, generator=None):
+        ans = torch.zeros(batch_size, rows, cols)
+    
+        for i in range(batch_size):
+            e_vals = torch.rand(min(rows, cols), generator=generator)
+            e_val_signs = torch.rand(min(rows, cols), generator=generator)
+            for j in range(len(e_vals)):
+                e_vals[j] *= -1 if (e_val_signs[j] < 0.5) else 1
+        
+            gaus = torch.randn (rows, rows, generator=generator)
+            svd = torch.linalg.svd (gaus)   
+            orth1 = svd[0]
+            orth1 = orth1[:, :min(rows, cols)]
+
+            gaus = torch.randn (cols, cols, generator=generator)
+            svd = torch.linalg.svd (gaus)   
+            orth2 = svd[2]
+            orth2 = orth2[:min(rows, cols), :]
+
+            ans[i, :, :] = torch.matmul(torch.matmul(orth1, torch.diag(e_vals)), orth2)
+            
+        return ans
+    
+    # Assume that we are instantiating A, B, and C matrices for the following
+    # state space model: 
+    #
+    #           x_k = A * x_(k-1) + B * u_k
+    #           y_k = C * x_k
+    #
+    # If x_k is (n_dims, 1), u_k is scalar, and y_k is (hidden_layer_size, 1),
+    # A -> (n_dims, n_dims), B -> (n_dims, 1), C -> (hidden_layer_size, n_dims)
+    @staticmethod
+    def generate_pool_dict(n_dims, num_tasks, curriculum=None, hidden_layer_size=100, **kwargs):
+        return {
+            "A": KalmanFilter.generate_rand_stable(num_tasks, n_dims, n_dims),
+            "B": KalmanFilter.generate_rand_stable(num_tasks, n_dims, n_dims),
+            "C": KalmanFilter.generate_rand_stable(num_tasks, 1, n_dims),
+            "x_k_1": KalmanFilter.generate_rand_stable(num_tasks, 1, n_dims)
+        }
+
+
+    @staticmethod
+    def get_metric():
+        return squared_error
+
+    @staticmethod
+    def get_training_metric():
+        return mean_squared_error
+
+class NoisyKalmanFilter(KalmanFilter):
+    def __init__(
+        self,
+        n_dims,
+        batch_size,
+        curriculum,
+        pool_dict=None,
+        seeds=None,
+        scale=1,
+        hidden_layer_size=100,
+    ):
+        """scale: a constant by which to scale the randomly sampled weights."""
+        super(NoisyKalmanFilter, self).__init__(n_dims, batch_size, curriculum, pool_dict, seeds)
+        self.scale = scale
+        self.hidden_layer_size = hidden_layer_size
+        self.curriculum = curriculum
```

Comment:
These parameters are already set by the `super(...).__init__(...)` call

Reply:
fixed this